src/ecs/resources/input/axis.rs

```diff
@@ -0,0 +1,11 @@
+use super::Button;
+
+/// Represents an axis made up of digital inputs, like W and S or A and D.
+/// Two of these could be analogous to a DPAD.
+#[derive(Serialize, Deserialize)]
+pub struct Axis {
+    /// Positive button, when pressed down axis value will return 1 if `neg` is not pressed down.
```

Comment:
Traditionally I had 3 types of inputs in my old system.

Boolean:  Simple on/off, like a button or a key.

Absolute:  Can be like a joystick axis or a button that is pressure sensitive, regardless they have defined min/max values (which can be queried for based on the type).

Relative:  Like a mouse axis, it can move an 'arbitrary' amount in a single call with no defined min or max.  I also exposed a 'scale' value with the type to allow for scaling based on user settings for per-hardware.

Do note, my system supported and distinguished between multiple keyboards, mice, and gamepads (and a Wiimote since I had one to test with) without issue (even on Windows, I used a low-level hook to get input instead of directinput, which let me distinguish between multiple input devices, on linux this was trivial).